common/sai_client/sai_thrift_client/sai_thrift_client.py

```diff
@@ -0,0 +1,122 @@
+import json
+import logging
+from functools import wraps
+from saichallenger.common.sai_client.sai_client import SaiClient
+from saichallenger.common.sai_client.sai_thrift_client.sai_thrift_utils import *
+from saichallenger.common.sai_data import SaiData
+from saichallenger.common.sai_data import SaiObjType
+from sai_thrift import sai_rpc, sai_adapter
+from thrift.protocol import TBinaryProtocol
+from thrift.transport import TSocket, TTransport
+
+
+def assert_status(method):
+    @wraps(method)
+    def method_wrapper(self, *args, do_assert=True, **kwargs):
+        try:
+            result = method(self, *args, **kwargs)
+        except Exception as e:
+            if do_assert:
+                raise AssertionError from e
+            else:
+                # TODO: pick correct STATUS
+                return sai_headers.SAI_STATUS_FAILURE
+        if do_assert and result is not None:
+            return result
+        else:
+            return sai_headers.SAI_STATUS_SUCCESS
+
+    return method_wrapper
+
+
+class SaiThriftClient(SaiClient):
+    """Thrift SAI client implementation to wrap low level SAI calls"""
+
+    def __init__(self, client_config):
+        self.thrift_transport = TSocket.TSocket(client_config['ip'], client_config['port'])
+        self.thrift_transport = TTransport.TBufferedTransport(self.thrift_transport)
+        protocol = TBinaryProtocol.TBinaryProtocol(self.thrift_transport)
+        self.thrift_transport.open()
+        self.thrift_client = sai_rpc.Client(protocol)
+
+    def __del__(self):
+        self.thrift_transport.close()
+
+    @assert_status
+    def create(self, obj_type, *, key=None, attrs=()):
+        oid_or_status = self._operate('create', attrs=attrs, obj_type=obj_type, key=key)
+        return oid_or_status if key is None else key
+
+    @assert_status
+    def remove(self, *, oid=None, obj_type=None, key=None):
```

Comment:
We should be able to get `obj_type` by `oid`. So, why do we need this extra param? Same for `key`..